api/admin.py

```diff
@@ -102,8 +103,29 @@ def get_listings(status: str = 'all'):
             code = 200
 
     # If NOT in admin session, deny access:
+    else:
+        response['err_msg'] = 'ACCESS DENIED.'
+
+    return response
+
+
+@admin.route('/set-listing/<id>-<status>', methods=['GET'])
+def action_on_listings(id: int, status: str):
+    """Route accepts a listing:
+
+    NOTE: must be in admin session.
+    """
+    response = dict()
+    # If in an admin session:
+    if 'username' in session:
+        if status in ['active', 'inactive', 'rejected', 'pending']:
+            listing = ListingsModel.query.get(id)
+            listing.status = status
+            db.session.commit()
+        else:
+            response['err_msg'] = status
+    # If NOT in admin session, deny access:
     else:
         response['err_msg'] = 'Access Denied.'
-        code = 403
 
-    return response, code
+    return response
```

Comment:
This will be an empty dict if I do not get an error. Can you add status code to signify if status change was successful and then maybe send me back the new status?

Reply:
Or even send me back the listing so I can see it has new status